backend/src/result_bool.rs

```diff
@@ -3,31 +3,32 @@
 #![deny(clippy::all, clippy::pedantic)]
 
 use qir_stdlib::strings::__quantum__rt__string_create;
-use std::ffi::{c_char, c_void, CString};
+use std::ffi::{CString, c_char, c_void};
 
-#[no_mangle]
+#[unsafe(no_mangle)]
 pub extern "C" fn __quantum__rt__result_get_zero() -> *mut c_void {
     std::ptr::null_mut()
 }
 
-#[no_mangle]
+#[unsafe(no_mangle)]
+#[allow(clippy::manual_dangling_ptr)]
 pub extern "C" fn __quantum__rt__result_get_one() -> *mut c_void {
     1 as *mut c_void
 }
 
-#[no_mangle]
+#[unsafe(no_mangle)]
 pub extern "C" fn __quantum__rt__result_equal(r1: *mut c_void, r2: *mut c_void) -> bool {
     r1 == r2
 }
 
-#[no_mangle]
+#[unsafe(no_mangle)]
```

Comment:
So just having the API be "pub extern C" requires it to be declared unsafe now, even if it doesn't do anything? :-/

Reply:
p.s. This is a different requirement. Here this is the requirement that the 'no_mangle' attribute be marked unsafe, not the extern (see https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-attributes.html).